sys/dev/drm/panfrost/panfrost_gem.c

```diff
@@ -320,11 +321,14 @@ panfrost_gem_fault(struct vm_area_struct *vma, struct vm_fault *vmf)
 			return (VM_FAULT_SIGBUS);
 	}
 
-	if (vm_page_busied(page))
-		goto fail_unlock;
-	if (vm_page_insert(page, obj, pidx))
+	if (!vm_page_tryxbusy(page)) {
+		vm_page_busy_sleep(page, "panfrost", 0);
+		goto retry;
+	}
+	if (vm_page_insert(page, obj, pidx)) {
```

Comment:
This might be easier to read as `if (vm_page_insert(...) != 0)` here and in other places, though that is an old bug.  Especially in contrast to the nearby bool return value of `vm_page_tryxbusy`.